torch/csrc/jit/register_prim_ops.cpp

```diff
@@ -1838,7 +1838,11 @@ int listSort<at::Tensor>(Stack& stack) {
   std::sort(
       list.begin(),
       list.end(),
-      [reverse](const at::Tensor& a, const at::Tensor& b) {
+      [reverse](const at::Tensor& a, const at::Tensor& b) -> bool {
+        // "strict weak ordering" issue - see other sort
+        if (a.getIntrusivePtr() == b.getIntrusivePtr()) {
+          return false;
+        }
         return (a.lt(b).is_nonzero()) ^ reverse;
```

Comment:
This code is pretty insane. Can't we just be explicit here:

```
if (reverse) {
  return a.gt(b).is_nonzero();
} else {
  return a.lt(b).is_nonzero();
}
```

Reply:
https://github.com/pytorch/pytorch/pull/19706#discussion_r279554924

who do i listen to lol

Reply:
my 2c: takes me longer to figure out what xor does than the control flow james has above

Reply:
The code James has above has a bug; `a.gt(b).is_nonzero() != (!a.lt(b).is_nonzero())`

I would personally prefer what I had in the original version `out = reverse ? !out : out`